src/Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests/FunctionalTests.cs

```diff
@@ -24,7 +24,49 @@ public static IEnumerable<object[]> GetDataForHl7v2()
         {
             var data = new List<string[]>
             {
+                new[] { @"ADT_A01", @"ADT_A01.hl7", @"ADT-A01-expected.json" },
```

Comment:
The functional tests fail for all the sample data and templates. Could you please double check about it? The tests could be found in `Visual Studio 2019 - Test Explorer - Microsoft.Health.Fhir.Liquid.Converter.FunctionalTests.FunctionalTests`.

Reply:
We have pushed the changes for the test case failure. There are 40 new cases which has been added and 13 which already present. It is expected that those 13 test case will fail as the template code has been changed.

We have added one condition to check null reference in FunctionalTest.cs as we are not creating "div" attribute in "Provenance" resource and while running the existing code it is giving null reference. So check this , we have added the condition.
`new List<string>
            {
                "$.entry[?(@.resource.resourceType=='Provenance')].resource.text.div",
            }.ForEach(path =>
            {
                JToken expectedObjectToken = expectedObject.SelectToken(path,false);
                if(expectedObjectToken != null) 
                {
                    expectedObject.SelectToken(path,false).Parent.Remove();
                }
                JToken actualObjectToken = actualObject.SelectToken(path,false);
                if(actualObjectToken != null)
                {
                    actualObject.SelectToken(path,false).Parent.Remove();
                }
            });`

Reply:
For "Provenance" resource, I saw that we are removing the "div" generation by commenting them out [here](https://github.com/microsoft/FHIR-Converter/blob/c13e23ded1bfef1bf75496b6690b0e55b5259084/data/Templates/Hl7v2/Resource/_Provenance.liquid#L27). Are we expecting to do so? The "div" here is designed to record the template information like template version. If we do not remove it, we would not need to add this check.

Reply:
If we have modified the existing templates, we can update the expected results of original test cases accordingly.

Reply:
For the "Provenance" resource , as per our SME's there were no mapping for "div" so we have commented that. As you mention that "div" was designed to record the template information, now we will add it the provenance resource created from MSH segment.
Provenance resource which are created from other segments like EVN will not have this "div". 
We will remove the null check condition from FunctionTests.cs which we have added.

We will update the expected result of original test cases once get the confirmation from Gina.

Reply:
@BoyaWu10 : We have added the "div" attribute to provenance and reverted the null condition check. We have updated the expected json for the 13 test case.